mewarpx/mewarpx/assemblies.py

```diff
@@ -32,6 +48,116 @@ def getvoltage(self):
         """Allows for time-dependent implementations to override this."""
         return mwxrun.eval_expression_t(self.V)
 
+    def getvoltage_e(self):
+        """Allows for time-dependent implementations to override this."""
+        return self.getvoltage() + self.WF
+
+    def init_scrapedparticles(self, fieldlist):
+        """Set up the scraped particles array. Call before
+        append_scrapedparticles.
+
+        Arguments:
+            fieldlist (list): List of string titles for the fields. Order is
+                important; it must match the order for future particle appends
+                that are made.
+        """
+        self._scrapedparticles_fields = fieldlist
+        self._scrapedparticles_data = AppendableArray(
+            typecode='d', unitshape=[len(fieldlist)])
+
+    def record_scrapedparticles(self):
+        """Handles transforming raw particle information from the WarpX
+        scraped particle buffer to the information used to record particles as
+        a function of time.
+
+        Note:
+            Assumes the fixed form of fields given in Assembly().  Doesn't
+            check since this is called many times.
+        """
```

Comment:
```suggestion

        Note:
            The total charge scraped and energy of the particles scraped 
            is multiplied by -1 since these quantities are leaving the system.
        """
```